src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,608 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.ConvertLinq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertLinq
+{
+    internal sealed class CSharpConvertLinqQueryToLinqMethodProvider : AbstractConvertLinqProvider
+    {
+        private static readonly TypeSyntax VarNameIdentifier = SyntaxFactory.IdentifierName("var");
+
+        protected override IAnalyzer CreateAnalyzer(SemanticModel semanticModel, CancellationToken cancellationToken)
+            => new CSharpAnalyzer(semanticModel, cancellationToken);
+
+        private sealed class CSharpAnalyzer : AnalyzerBase<QueryExpressionSyntax, SyntaxNode, SyntaxNode>
+        {
+            public CSharpAnalyzer(SemanticModel semanticModel, CancellationToken cancellationToken)
+                : base(semanticModel, cancellationToken)
+            {
+            }
+
+            protected override string Title => CSharpFeaturesResources.Convert_LINQ_query_to_foreach;
```

Comment:
title string should be "Convert_query_to_foreach"

Reply:
your previous comment was:
"resource string should match english.  so this should be Convert_LINQ_query_to_foreach."

:-)

---
In reply to: [176296860](https://github.com/dotnet/roslyn/pull/25362#discussion_r176296860) [](ancestors = 176296860)

Reply:
Discussed with @sharwell. Changed to Convert_to_foreach. Will merge with @cyrus.najmabadi@gmail.com and @heejaechang@outlook.com commits later.

---
In reply to: [176511113](https://github.com/dotnet/roslyn/pull/25362#discussion_r176511113) [](ancestors = 176511113,176510315,176296860)

Reply:
> :-)

Yes.  They should be *consistent* between each other.  So either one is fine, as long as they're consistent.  That said, i agree we should go with "Convert 'query' to 'foreach'" as the user string.  This will be consistent with the work that @heejaechang and i are doing.